unit/java_bytecode/java_bytecode_parse_generics/parse_generic_class_with_generic_inner_classes.cpp

```diff
@@ -0,0 +1,918 @@
+
+    THEN("The inputs are of correct type")
+    {
+      const auto &param_type =
+        require_type::require_parameter(function_call, "input");
+      require_type::require_pointer(
+        param_type.type(), symbol_typet(class_prefix + "$InnerClass"));
+      require_type::require_java_generic_type(
+        param_type.type(),
+        {{require_type::type_parameter_kindt::Var, class_prefix + "::T"}});
+    }
+  }
+  THEN(
+    "Method 2 should take two pointers to "
+    "GenericClassWithGenericInnerClasses$InnerClass")
+  {
+    const std::string func_name = ".method2";
+    const std::string func_descriptor =
+      ":(LGenericClassWithGenericInnerClasses$InnerClass;"
+      "LGenericClassWithGenericInnerClasses$InnerClass;)V";
+    const std::string process_func_name =
+      class_prefix + func_name + func_descriptor;
+
+    REQUIRE(new_symbol_table.has_symbol(process_func_name));
+    const symbolt &function_symbol =
+      new_symbol_table.lookup_ref(process_func_name);
+
+    const code_typet &function_call =
+      require_type::require_code(function_symbol.type, 3);
+
+    THEN("The inputs are of correct type")
+    {
+      const auto &param_type =
+        require_type::require_parameter(function_call, "input");
+      require_type::require_pointer(
+        param_type.type(), symbol_typet(class_prefix + "$InnerClass"));
+      require_type::require_java_generic_type(
+        param_type.type(),
+        {{require_type::type_parameter_kindt::Var, class_prefix + "::T"}});
+
+      const auto &param_type2 =
+        require_type::require_parameter(function_call, "input2");
+      require_type::require_pointer(
+        param_type2.type(), symbol_typet(class_prefix + "$InnerClass"));
+      require_type::require_java_generic_type(
+        param_type2.type(),
+        {{require_type::type_parameter_kindt::Var, class_prefix + "::T"}});
+    }
+  }
+  THEN(
+    "Method 3 should take a pointer to "
+    "GenericClassWithGenericInnerClasses$GenericInnerClass")
+  {
+    const std::string func_name = ".method3";
+    const std::string func_descriptor =
+      ":(LGenericClassWithGenericInnerClasses$GenericInnerClass;)V";
+    const std::string process_func_name =
+      class_prefix + func_name + func_descriptor;
+
+    REQUIRE(new_symbol_table.has_symbol(process_func_name));
+    const symbolt &function_symbol =
+      new_symbol_table.lookup_ref(process_func_name);
+
+    const code_typet &function_call =
+      require_type::require_code(function_symbol.type, 2);
+
+    THEN("The inputs are of correct type")
+    {
+      const auto &param_type =
+        require_type::require_parameter(function_call, "input");
+      require_type::require_pointer(
+        param_type.type(), symbol_typet(class_prefix + "$GenericInnerClass"));
+      require_type::require_java_generic_type(
+        param_type.type(),
+        {{require_type::type_parameter_kindt::Var, class_prefix + "::T"},
```

Comment:
Very minor nit: Again, would be nice to keep the formatting of the initializer list consistent for all elements